Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/symptoms/calendar/SubmissionSymptomCalendarViewModel.kt

```diff
@@ -92,29 +83,29 @@ class SubmissionSymptomCalendarViewModel @AssistedInject constructor(
         performSubmission {
             analyticsKeySubmissionCollector.reportSubmittedAfterSymptomFlow(testType)
         }
+        routeToScreen.postValue(
+            SubmissionSymptomCalendarFragmentDirections
+                .actionSubmissionSymptomCalendarFragmentToSubmissionDoneFragment(testType)
+        )
     }
 
     fun onCancelConfirmed() {
         Timber.d("onCancelConfirmed() clicked on calendar screen.")
         performSubmission {
             analyticsKeySubmissionCollector.reportSubmittedAfterCancel(testType)
         }
+        routeToScreen.postValue(
+            SubmissionSymptomCalendarFragmentDirections.actionSubmissionSymptomCalendarFragmentToMainFragment()
```

Comment:
This should not return to the main fragment, but to the tab from where the UQS was opened. I've also opened a PR that should fix this navigation issue within the submission flow: #5408

Reply:
We may not have passed the UQS, eg. if we enter the submission flow from home screen on the TAN flow

Reply:
Yep, let's just merge this one and I'll address it in my PR.